src/os/ui/datetime/datecontrol.js

```diff
@@ -262,9 +331,17 @@ os.ui.datetime.DateControlCtrl.prototype.updateController_ = function() {
  * @export
  */
 os.ui.datetime.DateControlCtrl.prototype.shiftDate = function(direction) {
+  var modifier = 1;
   if (!this['disabled']) {
-    this['startDate'] = os.time.offset(this['startDate'], this['duration'], direction, true);
-    this['endDate'] = os.time.offset(this['endDate'], this['duration'], direction, true);
+    if (this['duration'] === os.time.Duration.CUSTOM) {
+      // For custom durations, multiply the offset by the difference in days between the start and end dates
+      const millisecondsPerDay = 1000 * 60 * 60 * 24;
```

Comment:
Nit: we have a constant for this (and many similar static durations) in time.js: `os.time.millisecondsInDay`.